### PR TITLE
[TASK] Add List of PSR-14 Events

### PR DESCRIPTION
**Pull request by susannemoog**
All events should be documented right from the start (to avoid the search, trial and error flaws of hooks and signals).

### CODE REVIEW

File: Documentation/ApiOverview/Hooks/Events/Workspaces/AfterCompiledCacheableDataForWorkspaceEvent.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   :Description:
+         Set / Overwrite the cacheable data of versions.
+   :ReturnType:
+         void
+
+
+ - :Method:
+         getVersions()
+   :Description:
+         Get versions.
```
Comment (ervaude) [inline Documentation/ApiOverview/Hooks/Events/Workspaces/AfterCompiledCacheableDataForWorkspaceEvent.rst]: Returns

File: Documentation/ApiOverview/Hooks/Events/Workspaces/SortVersionedDataEvent.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+   :Description:
+         Set / Overwrite column to sort by.
+   :ReturnType:
+         string
+
+
+ - :Method:
+         getSortDirection()
+   :Description:
+         Get the current sorting direction (ASC | DESC)
```
Comment (ervaude) [inline Documentation/ApiOverview/Hooks/Events/Workspaces/SortVersionedDataEvent.rst]: Returns

File: Documentation/ApiOverview/Hooks/Events/Workspaces/SortVersionedDataEvent.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+   :Description:
+         Set / Overwrite the cacheable data of versions.
+   :ReturnType:
+         void
+
+
+ - :Method:
+         getSortColumn()
+   :Description:
+         Get column to sort by.
```
Comment (ervaude) [inline Documentation/ApiOverview/Hooks/Events/Workspaces/SortVersionedDataEvent.rst]: Returns

File: Documentation/ApiOverview/Hooks/Events/Workspaces/GetVersionedDataEvent.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+   :Description:
+         Get current start index.
+   :ReturnType:
+         int
+
+
+ - :Method:
+         getLimit()
+   :Description:
+         Get current limit / maximum number of data sets.
```
Comment (ervaude) [inline Documentation/ApiOverview/Hooks/Events/Workspaces/GetVersionedDataEvent.rst]: Returns

File: Documentation/ApiOverview/Hooks/Events/Workspaces/GetVersionedDataEvent.rst [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+   :Description:
+         Set / overwrite currently relevant (with applied limit) part of the data array.
+   :ReturnType:
+         void
+
+
+ - :Method:
+         getStart()
+   :Description:
+         Get current start index.
```
Comment (ervaude) [inline Documentation/ApiOverview/Hooks/Events/Workspaces/GetVersionedDataEvent.rst]: Returns

File: Documentation/ApiOverview/Hooks/Events/Workspaces/GetVersionedDataEvent.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+   :Description:
+         Set / Overwrite the cacheable data of versions.
+   :ReturnType:
+         void
+
+
+ - :Method:
+         getDataArrayPart()
+   :Description:
+         Get currently relevant (with applied limit) part of the data array.
```
Comment (ervaude) [inline Documentation/ApiOverview/Hooks/Events/Workspaces/GetVersionedDataEvent.rst]: Returns

File: Documentation/ApiOverview/Hooks/Events/Workspaces/AfterDataGeneratedForWorkspaceEvent.rst [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+   :Description:
+         Set / Overwrite the cacheable data of versions.
+   :ReturnType:
+         void
+
+
+ - :Method:
+         getVersions()
+   :Description:
+         Get versions.
```
Comment (ervaude) [inline Documentation/ApiOverview/Hooks/Events/Workspaces/AfterDataGeneratedForWorkspaceEvent.rst]: Returns

File: Documentation/ApiOverview/Hooks/EventDispatcher/Events/Workspaces/AfterDataGeneratedForWorkspaceEvent.rst [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+   :Description:
+         Returns the current instance of the GridDataService.
+   :ReturnType:
+         \TYPO3\CMS\Workspaces\Service\GridDataService
+
+
+ - :Method:
+         getData()
+   :Description:
+         Returns the cacheable data of versions.
```
[the remark this answers is not in the document]
Comment (susannemoog) [reply Documentation/ApiOverview/Hooks/EventDispatcher/Events/Workspaces/AfterDataGeneratedForWorkspaceEvent.rst]: yes.

**Pull request merged**